src/Controller/SentencesListsController.php

```diff
@@ -365,6 +367,28 @@ public function add_new_sentence_to_list()
         $this->set('sentence', $result);
     }
 
+    public function add_sentence_to_new_list() {
+        $userId = $this->Auth->user('id');
+        $sentenceId = $this->request->getData('sentenceId');
+        $list = $this->SentencesLists->createList(
+            $this->request->getData('name'), 
+            $this->Auth->user('id')
+        );
+
+        if ($this->SentencesLists->addSentenceToList($sentenceId, $list->id, $userId)) {
+            $list->hasSentence = true;
+            $this->set('result', $list);
+            $this->Cookie->write('most_recent_list', $list->id, false, '+1 month');
+        } else {
+            $this->set('result', 'error');
+        }
+        
+        $this->loadComponent('RequestHandler');
+        $this->set('result', $list);
```

Comment:
I think this line should be removed.

You've already set `result` above (on line 380 and 383) and in case of an error, this line would overwrite the error response and would show the sentence as added to the list although it wasn't.

Reply:
Good catch. Thanks!